lets-revise-hooks-today.md

```diff
@@ -0,0 +1,656 @@
+Hello Everyone
+
+I'm am Minhaj Ahmad Khan and I'm a Front-End developer. After a good response to my first blog. Thanks for your like and support so here I am back with a new blog I am sure you like it.
+
+## Why Hooks?
+
+Until now if you started writing a functional component and ran into a situation where you needed to add a state you would have to convert the component to a class component that is simply because the state could only be used in class components That where Hooks comes into the picture.
+
+## What is Hooks?
+
+A Hook is a special function that lets you “hook into” React features. For example, useState is a Hook that lets you add React state to function components.
+
+So In this blog, we cover all major Hooks where we cover everything about these Hooks in depth.
+
+I am assuming readers have intermediate knowledge of React.
+
+So what we are going to cover
+
+## Rules of Hooks
+
+- Only call Hooks at the top level. Don’t call Hooks inside loops, conditions, or nested functions.
+
+- Only call Hooks from React function components. Don’t call Hooks from regular JavaScript functions. (There is just one other valid place to call Hooks — your own custom Hooks. We’ll learn about them in a moment.)
+
+## Basic Hooks
+
+- useState
+
+- useEffect
+
+- useContext
+
+## Additional Hooks
+
+- useReducer
+
+- useCallback
+
+- useMemo
+
+### State Hook (useState)
+
+State Hook(useState) is a hook that let you add react state to functional components.
```

Comment:
Talk about how to use `useState`. Example : `useState` takes initial state as an argument and returns a tuple of state and and updater function. We can use this updater function to update the state. The order is important here